docs/how-to/enable_hugepages_for_upf.md

```diff
@@ -0,0 +1,68 @@
+# Enable HugePages for UPF
+
+## Requirements
+
+- A Kubernetes host with a CPU supporting AVX2 and RDRAND instructions (Intel Haswell, AMD Excavator or equivalent)
+- A Kubernetes cluster with the Multus addon enabled
+- Juju = 3.1/stable
+- HugePages is enabled in Kubernetes Host and it has enough 2Gi free HugePages 
+- Kubernetes Host CPU should have the flag "pdpe1gb" which allows to set 1Gi HugePages
+
+## Deploy the UPF
+
+Deploy the UPF charm in a Juju model named `user-plane`:
+
+```bash
+juju add-model user-plane
+juju deploy sdcore-upf --trust --channel=edge
+```
+
+## Enable HugePages
+
+```bash
+juju config sdcore-upf enable-hugepages=True
+```
+
+Container resources are hardcoded with 2 CPU and HugePage resources (requests:hugepages-1Gi, limits:2Gi).
+
+```yaml
+limits:
+  cpu: "2"
+  hugepages-1Gi: 2Gi
+requests:
+  cpu: "2"
+  hugepages-1Gi: 2Gi
+```
+
+Check the bessd container logs which indicates that hugepages are reserved:
+```bash
+kubectl logs -f  sdcore-upf-0 -c  bessd -n user-plane  | grep -i huge
+# Sample Output
+2023-10-26T12:33:35.495Z [bessd] I1026 12:33:35.495568    54 dpdk.cc:169] Initializing DPDK EAL with options: ["bessd", "--main-lcore", "127", "--lcore", "127@0-31", "--no-shconf", "--legacy-mem", "--socket-mem", "1024", "--huge-unlink"]
+2023-10-26T12:33:35.506Z [bessd] EAL: 48967 hugepages of size 2097152 reserved, but no mounted hugetlbfs found for that size
+```
+
+Besides, PacketPool is crated with name`DpdkPacketPool` in bessd container logs if HugePages are enabled:
+```bash
+kubectl logs -f  sdcore-upf-0 -c  bessd -n user-plane  | grep -i pool
+# Sample Output
+2023-10-26T12:33:35.705Z [bessd] I1026 12:33:35.705854    54 packet_pool.cc:49] Creating DpdkPacketPool for 262144 packets on node 0
+```
+
+## Disable HugePages
+
+```bash
+juju config sdcore-upf-operator enable-hugepages=False
+```
+Check the bessd container logs and no logs appears with hugepages reservation. 
+```bash
+kubectl logs -f  sdcore-upf-0 -c  bessd -n user-plane  | grep -i huge
+# No Output
+```
+
+PacketPool name appears as `PacketPool0` in bessd container logs.
+```bash
+gatici@gaticipc:~$ kubectl logs -f  sdcore-upf-0 -c  bessd -n user-plane | grep -i Pool
```

Comment:
Avoid using your name in the examples, try to use `user@ubuntu` or something along those lines

Reply:
Done